### ISSUE TITLE
Connect audit recommendations with CUBE

### ISSUE THREAD

**Issue by joaocarloscardoso** [joaocarloscardoso/AITAM#81]
At audit recommendation matrix level connect with cube.
Several possibilities:
- Send recommendation description and/or related findings to CUBE end point and (assyncronously) receive through NLP (and using GATE) descriptive keywords and similar cases
- Process recommendation description and/or related findings through NLP techniques (at AITAM level), and, based on results, search the XML cube and gazeteers for keywords and similar cases.

Activated through push button to render results and similarities

**Comment by joaocarloscardoso** [comment]
Prototypes were developed and integrated currently as differente repositories in GitHub:
-  [Client request and notification](https://github.com/joaocarloscardoso/aitam_notclient1);
- [Listener, responsible to address the CUBE and maitain requests status](https://github.com/joaocarloscardoso/aitam_notification).

**Comment by pawelbanas** [comment]
On the CUBE side, the semantic analysis can already be performed locally: a .txt or .docx is analysed in an NLP way and similar profile is searched through the original text's senetnces. Refinement of the search is foreseen. There is lack of the net communication.

**Comment by joaocarloscardoso** [comment]
True. A prototype of the communications module was developed as a separated project so far (less entropy :) )
The idea of connection with CUBE relies on a asyncronous notification system based on a REST API
